[ossaris56] iP

ossaris56 opened the pull request:
# Hello! This is my heading.
- And this
- Is my
- Bullet list

1. A number
2. List
3. Too

```python
def hello():
    print('This is my fenced code block (with syntax highlighting)')
```

- [x] Task 1
- [ ] Task 2
- [ ] Task 3 and emoji 😣 

> My blockquote and a [hyperlink](https://cameo.jerma.io/)

An inline code : `echo hello`

Finally **bold**, *italic* and ~~strikethrough~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ossaris56 committed:
Renamed chatbot, add greet and exit

ossaris56 committed:
add echo

ossaris56 committed:
add store user input

ossaris56 committed:
add mark and unmark as done

ossaris56 committed:
refactored to use switch instead of if and added classes

ossaris56 committed:
add todos, events, deadlines function

ossaris56 committed:
add automated text ui testing

ossaris56 committed:
add exceptions

ossaris56 committed:
add delete function

ossaris56 committed:
add create data file

ossaris56 committed:
Add writing to file when tasks change

ossaris56 committed:
Add load tasklist on startup

ossaris56 committed:
Merge branch 'branch-Level-7'

ossaris56 committed:
Add dates and times processing

ossaris56 committed:
Merge branch 'branch-Level-8'

ossaris56 committed:
Refactor code into classes

ossaris56 committed:
add command classes and refactor ui tasklist and storage

ossaris56 committed:
add parse in parser

ossaris56 committed:
Implement execute in ListCommand

ossaris56 committed:
Implement execute in MarkCommand

ossaris56 committed:
implement execute in DeadlineCommand and EventCommand

ossaris56 committed:
Implement execute in DeleteCommand

ossaris56 committed:
refactor main and bugfix unmarking task

ossaris56 committed:
Merge branch 'A-MoreOOP'

ossaris56 committed:
Add packages

ossaris56 committed:
Merge branch 'A-Packages'

ossaris56 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ossaris56 committed:
add gradle

ossaris56 committed:
Add JUnit tests

ossaris56 committed:
Merge branch 'A-JUnit'

ossaris56 committed:
fix bye bug and add building jar

ossaris56 committed:
Add JavaDoc to command classes

ossaris56 committed:
Add JavaDoc to exceptions

ossaris56 committed:
Add JavaDoc to task and missing fullstops in commands and exceptions

ossaris56 committed:
Add JavaDoc to parser

ossaris56 committed:
Add missing fullstop in parser and JavaDoc in storage

ossaris56 committed:
Add JavaDoc to Asura and Ui

ossaris56 committed:
Reformat parser and storage to have no indents on switch

ossaris56 committed:
Reformat spacing

ossaris56 committed:
Add find function

ossaris56 committed:
Merge branch 'branch-A-JavaDoc'

ossaris56 committed:
Merge branch 'branch-A-CodingStandard'

ossaris56 committed:
Merge branch 'branch-Level-9'

ossaris56 committed:
Add JavaDoc for FindCommand and TaskList

wuzengfu commented on src/main/java/asura/parser/Parser.java:
```diff
@@ -0,0 +1,83 @@
+package asura.parser;
+
+import asura.commands.*;
```
imported classes should be listed explicitly

wuzengfu commented on src/main/java/asura/data/tasks/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+
+    /**
+     * Creates a new list of tasks with the specified tasks
+     * @param taskList The tasks specified
```
missing fullstop at the end of the parameter description
```java
@param taskList The tasks specified.
```

wuzengfu commented on src/main/java/asura/Asura.java:
```diff
@@ -0,0 +1,64 @@
+
+    /**
+     * Executes the main loop
+     * @param args Optional arguments passed to the program
```
missing full stop at the end of the description

wuzengfu commented on src/main/java/asura/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,57 @@
+     * Indicates that the user does not want to terminate the program.
+     * @return A boolean representing whether the program is to be terminated, in this case false.
+     */
+    public boolean isExit() {
```
although this does not violate coding standard, it is good to have a `@Override` annotation since it overrides `Command::isExit`

wuzengfu reviewed (commented):
LGTM. I noted some minor coding standard violations as comments . Overall, the code is easy to read, good structure of the code and use of `StringBuilder` to handle string concatenation!

nws321 commented on src/main/java/asura/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,57 @@
+     * @param dateArray An array of the date specified by the user.
+     */
+    public DeadlineCommand(List<String> descriptionArray, List<String> dateArray) {
+        this.descriptionArray = descriptionArray;
```
Could use "descriptions" as variable name instead to shorten it

nws321 commented on src/main/java/asura/data/tasks/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm:ss a")) + ")";
```
Could split line to new line as line seems slightly long

nws321 commented on src/main/java/asura/data/tasks/Event.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + start + " to: " + end + ")";
```
Could split line to new line as line seems slightly long

nws321 commented on src/main/java/asura/data/tasks/Task.java:
```diff
@@ -0,0 +1,48 @@
+        isDone = false;
+    }
+
+    public String toString() {
```
Similar to what zengfu mentioned, not required but good to have `@Override` annotation

nws321 commented on src/main/java/asura/data/tasks/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+        return taskList;
+    }
+
+    public String toString() {
```
... same `@Override` suggestion

nws321 commented on src/main/java/asura/data/tasks/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+     * Removes a task in the task list at the specified index.
+     * @param index The index of the task to be removed.
+     */
+    public Task removeAt(int index) {
```
Could put just "remove"

nws321 commented on src/main/java/asura/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+     * @return The saved list of tasks of the user.
+     * @throws AsuraException If loading the data fails.
+     */
+    public List<Task> load() throws AsuraException {
```
Could write "loadTask" instead for greater clarity

nws321 commented on src/main/java/asura/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+     * @param taskList The list of tasks to be saved.
+     * @throws AsuraException If saving the tasks fails.
+     */
+    public void save(List<Task> taskList) throws AsuraException {
```
... same as earlier

nws321 commented on src/test/java/asura/commands/DeadlineCommandTest.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    @Test
+    public void testExecute() throws AsuraException {
```
Could change to "exectuteTest" instead to make it a verb

nws321 commented on src/test/java/asura/commands/TodoCommandTest.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Test
+    public void testExecute() throws AsuraException {
```
... same as before

nws321 commented on src/main/java/asura/parser/Parser.java:
```diff
@@ -0,0 +1,83 @@
+        String taskString = String.join(" ", splitCommand.subList(1, splitCommand.size()));
+        List<String> descriptionArray;
+
+        switch (prefix) {
```
Think it is great that you use switch case as it is more efficient in this case

nws321 reviewed (commented):
I think is great you have so many clear classes for each component so that no class is over cluttered.

ossaris56 committed:
Add basic structure of GUI

ossaris56 committed:
Add chatbot replies to GUI

ossaris56 committed:
Add rendering chatbot response on the GUI

ossaris56 committed:
Merge branch 'branch-Level-10'

ossaris56 committed:
Add assertions

The current codebase does not currently check for non-empty data and
command elements, which can lead to unexpected errors or behavior when
processing input.

This needs to change to ensure that key assumptions hold true and to
make the code easier to debug

Add assertions to validate that variables such as splitCommand are not
empty. These assertions ensure that the program behaves as expected and
catches potential errors.

Assertions are used to verify assumptions during the runtime of the
program.

ossaris56 committed:
Merge pull request #1 from ossaris56/branch-A-Assertions

Add assertions

ossaris56 committed:
Remove dead code

ossaris56 committed:
Remove unused code in Asura and Ui

ossaris56 committed:
Merge pull request #2 from ossaris56/branch-A-CodeQuality

Improve code quality

ossaris56 committed:
Add help command

ossaris56 committed:
Add help command

ossaris56 committed:
Fixed find JavaDoc and accessor in MarkCommand

ossaris56 committed:
Merge branch 'branch-BCD-Extension' of https://github.com/ossaris56/ip into branch-BCD-Extension

ossaris56 committed:
Merge pull request #3 from ossaris56/branch-BCD-Extension

Add C-Help and C-BetterSearch

ossaris56 committed:
Add FXML

ossaris56 committed:
Add JavaDoc to initialize

ossaris56 committed:
Add JavaDoc to setAsura

ossaris56 committed:
Delete dead code

ossaris56 committed:
Fix JavaDoc in Storage

ossaris56 committed:
Fix JavaDoc in ByeCommand

ossaris56 committed:
Add title to GUI

ossaris56 committed:
Remove dead code

ossaris56 committed:
Add UI screenshot

ossaris56 committed:
Add user guide

ossaris56 committed:
Change invalid input message

ossaris56 committed:
Add error handling

ossaris56 committed:
Fix header comment in ByeCommand

ossaris56 committed:
Fix long method in Parser

ossaris56 committed:
Fix long method in Storage

ossaris56 committed:
Add JavaDoc to Parser and Storage
